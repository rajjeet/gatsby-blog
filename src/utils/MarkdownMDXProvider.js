import React from 'react';
import { MDXProvider } from '@mdx-js/react';
import { MDXRenderer } from 'gatsby-plugin-mdx';
import AutoLinkHeader from '../components/auto-link-header';

class MarkdownMdxProvider extends React.Component {
  state = {
    ...['h1', 'h2', 'h3', 'h4', 'h5', 'h6'].reduce((a, v) => {
      a[v] = AutoLinkHeader(v); // eslint-disable-line no-param-reassign
      return a;
    }, {}),
  };

  render() {
    const { content } = this.props;
    return (
      <MDXProvider components={this.state}>
        {content && <MDXRenderer>{content}</MDXRenderer>}
      </MDXProvider>
    );
  }
}

export default MarkdownMdxProvider;
