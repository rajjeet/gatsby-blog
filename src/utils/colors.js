export default {
    primaryColor: '#2185d0',
    secondaryColor: '#f3711c',
};

export const computerBreakpoint = '1023px';
export const tabletBreakpoint = '768px';
export const bigMobileBreakpoint = '425px';
export const smallMobileBreakpoint = '320px';
export const breakpoint600 = '600px';
export const boxShadow = '0 4px 8px 0 rgba(0,0,0,0.2)';
export const hoverBoxShadow = '0 4px 9px 0 rgba(0,0,0,0.6)';
export const lightBoxShadow = '0 4px 4px 0 rgba(0,0,0,0.15)';
export const primaryColor = '#2185d0';