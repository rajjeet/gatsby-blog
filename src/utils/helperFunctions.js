const _ = require(`lodash`);

module.exports = {
    getTagSlug: function (tag) {
        return `/tags/${_.kebabCase(tag)}/`
    }
};

