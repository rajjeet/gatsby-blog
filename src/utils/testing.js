export const createMockGatsbyImageSharpFluid = {
  file: {
    childImageSharp: {
      fluid: {
        base64: '',
        aspectRatio: 1,
        src: '',
        srcSet: '',
        sizes: '',
      },
    },
  },
};

export const createMockGroups = {
  categoryGrouping: {
    group: [
      {
        fieldValue: 'Experience',
        totalCount: 1,
      },
      {
        fieldValue: 'Guide',
        totalCount: 2,
      },
    ],
  },
};

export const createMockProjects = [
  {
    node: {
      frontmatter: {
        title: 'Condo Landing Pages',
        description: 'An automated process for building fast, responsive landing pages for pre-construction condos.',
        tags: null,
        thumbnail: createMockGatsbyImageSharpFluid.file,
      },
      fields: {
        slug: '/projects/condo-landing-pages/',
      },
    },
  },
  {
    node: {
      frontmatter: {
        title: 'Kitchen Quoter',
        description: 'A do-it-yourself calculator that generates dollar quotes based on user-specified fields and formula.',
        tags: null,
        thumbnail: createMockGatsbyImageSharpFluid.file,
      },
      fields: {
        slug: '/projects/kitchen-quoter/',
      },
    },
  },
];

export const createMockProjectLinks = [
  {
    label: 'ex. CharismaCondos.net',
    value: 'https://dgftd2qqywjy1.cloudfront.net',
  },
  {
    label: 'ex. SouthWoodbridge.com',
    value: 'https://d1n2vqbqyes277.cloudfront.net',
  },
  {
    label: 'Github Code',
    value: 'https://github.com/rajjeet/charismacondos',
  },
];

export const createMockPosts = [
  {
    node: {
      id: '13cd14f6-36e9-55cd-8a4e-4bf1aea2c9e6',
      timeToRead: 2,
      frontmatter: {
        title: 'Adding a Project Section to My Website',
        date: '11 May, 2019',
        tags: [
          'Gatsby JS X',
        ],
        category: 'Side Project X',
        description: "Over the past year, I truly learned the importance of building side projects as a software developer. Besides building projects, showing them to the world is just as important. That's why I created the project section.",
        image: createMockGatsbyImageSharpFluid.file,
      },
      fields: {
        slug: '/adding-project-section-to-website/',
      },
      excerpt: 'Original Intent My original intent was to connect related blog posts using…',
    },
  },
  {
    node: {
      id: '5f2933ac-25ed-5ae0-84d7-c5d59fc4448e',
      timeToRead: 4,
      frontmatter: {
        title: 'State Management using React Hooks',
        date: '02 May, 2019',
        tags: [
          'React X',
          'Angular X',
        ],
        category: 'Guide X',
        description: 'I was learning about the new React Hooks, when I came across a blog post that claims to build state management, like Redux, using React Hooks. I decided to try it out and found it surprisingly simple and effective.',
        image: createMockGatsbyImageSharpFluid.file,
      },
      fields: {
        slug: '/state-management-react-hooks/',
      },
      excerpt: 'Introduction This blog post is inspired by \n another post , that demonstrates…',
    },
  },
];
