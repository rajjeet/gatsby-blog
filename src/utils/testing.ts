export const createMockGatsbyImageSharpFluid = {
  file: {
    childImageSharp: {
      fluid: {
        base64: '',
        aspectRatio: 1,
        src: 'https://via.placeholder.com/250x180',
        srcSet: '',
        sizes: '',
      },
    },
  },
};

export const createMockGroups = {
  categoryGrouping: {
    group: [
      {
        fieldValue: 'Experience',
        totalCount: 1,
      },
      {
        fieldValue: 'Guide',
        totalCount: 2,
      },
    ],
  },
};

export const createMockProjects = [
  {
    node: {
      frontmatter: {
        title: 'Condo Landing Pages',
        description: 'An automated process for building fast, responsive landing pages for pre-construction condos.',
        thumbnail: createMockGatsbyImageSharpFluid.file,
        techStackTags: [
          {
            type: 'AWS',
            label: 'AWS CloudFormation',
          },
        ],
        links: [
          {
            label: 'Github Code',
            value: 'https://github.com/rajjeet/charismacondos',
          },
        ],
      },
      fields: {
        slug: '/projects/condo-landing-pages/',
      },
      body: null,
    },
  },
  {
    node: {
      frontmatter: {
        title: 'Kitchen Quoter',
        description: 'A do-it-yourself calculator that generates dollar quotes based on user-specified fields and formula.',
        techStackTags: [
          {
            type: 'JavaScript',
            label: 'GatsbyJS',
          },
          {
            type: 'AWS',
            label: 'AWS S3',
          },
        ],
        links: [
          {
            label: 'ex. CharismaCondos.net',
            value: 'https://dgftd2qqywjy1.cloudfront.net',
          },
          {
            label: 'ex. SouthWoodbridge.com',
            value: 'https://d1n2vqbqyes277.cloudfront.net',
          },
        ],
        thumbnail: createMockGatsbyImageSharpFluid.file,
      },
      fields: {
        slug: '/projects/kitchen-quoter/',
      },
      body: null,
    },
  },
];

export const createMockProjectLinks = [
  {
    label: 'ex. CharismaCondos.net',
    value: 'https://dgftd2qqywjy1.cloudfront.net',
  },
  {
    label: 'ex. SouthWoodbridge.com',
    value: 'https://d1n2vqbqyes277.cloudfront.net',
  },
  {
    label: 'Github Code',
    value: 'https://github.com/rajjeet/charismacondos',
  },
];

export const createMockPosts = [
  {
    node: {
      id: '13cd14f6-36e9-55cd-8a4e-4bf1aea2c9e6',
      timeToRead: 2,
      frontmatter: {
        title: 'Adding a Project Section to My Website',
        dateCreated: '11 May, 2019',
        dateModified: null,
        tags: [
          'Gatsby JS X',
        ],
        description: "Over the past year, I truly learned the importance of building side projects as a software developer. Besides building projects, showing them to the world is just as important. That's why I created the project section.",
        image: createMockGatsbyImageSharpFluid.file,
      },
      fields: {
        slug: '/adding-project-section-to-website/',
      },
      excerpt: 'Original Intent My original intent was to connect related blog posts using…',
      body: null,
    },
  },
  {
    node: {
      id: '5f2933ac-25ed-5ae0-84d7-c5d59fc4448e',
      timeToRead: 4,
      frontmatter: {
        title: 'State Management using React Hooks',
        dateCreated: '02 May, 2019',
        dateModified: null,
        tags: [
          'React X',
          'Angular X',
        ],
        description: 'I was learning about the new React Hooks, when I came across a blog post that claims to build state management, like Redux, using React Hooks. I decided to try it out and found it surprisingly simple and effective.',
        image: createMockGatsbyImageSharpFluid.file,
      },
      fields: {
        slug: '/state-management-react-hooks/',
      },
      excerpt: 'Introduction This blog post is inspired by \n another post , that demonstrates…',
      body: null,
    },
  },
  {
    node: {
      id: 'ea9849ff-7e43-5fe2-88d8-5c441cfb4dad',
      timeToRead: 3,
      frontmatter: {
        title: 'Build A Super Customized and Performant Blog',
        dateCreated: '21 February, 2019',
        dateModified: null,
        tags: [
          'Blog',
          'React',
        ],
        description: 'Starting a blog seemed easy to me, but selecting the right tool was not straight forward for my goals. See why I decided to use Gatsby to host Ortmesh, and how you might benefit from doing the same.',
        image: createMockGatsbyImageSharpFluid.file,
      },
      fields: {
        slug: '/build-super-customized-performant-blog/',
      },
      excerpt: 'Humble Beginnings For most of us, a blog is about sharing ideas, helping others…',
      body: null,
    },
  },
];

export const mockPost = {
  id: '5f2933ac-25ed-5ae0-84d7-c5d59fc4448e',
  timeToRead: 4,
  frontmatter: {
    title: 'State Management using React Hooks',
    dateCreated: '02 May, 2019',
    dateModified: null,
    tags: [
      'React X',
      'Angular X',
    ],
    category: 'Guide X',
    description: 'I was learning about the new React Hooks, when I came across a blog post that claims to build state management, like Redux, using React Hooks. I decided to try it out and found it surprisingly simple and effective.',
    image: {
      publicURL: '/static/rawpixel-780494-unsplash-e36954fd765f72a96646fa7c3fa89d49.jpg',
    },
  },
  fields: {
    slug: '/state-management-react-hooks/',
  },
  excerpt: 'Introduction This blog post is inspired by \n another post , that demonstrates…',
  body: null,
  tableOfContents: {
    items: [
      {
        url: '#overview',
        title: 'Overview',
      },
      {
        url: '#the-why',
        title: 'The Why',
      },
    ],
  },
};
