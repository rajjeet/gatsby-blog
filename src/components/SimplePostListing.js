import {Link} from "gatsby";
import React from "react";
import styled from 'styled-components';
import * as theme from '../utils/colors';

const SimplePostListing = ({posts, className}) => (
    <div className={className}>
        <h3>Blog Posts</h3>
        <ul>
            {
                posts.map(({node}) => (
                    <li>
                        <S.Link to={node.fields.slug}>
                            <h4>{node.frontmatter.title}</h4>
                            <div>
                                <span>{node.frontmatter.date}</span>
                                <span>&emsp;{node.timeToRead} mins read</span>
                            </div>
                        </S.Link>
                    </li>
                ))
            }
        </ul>
    </div>
);

const S = {
    Link: styled(Link)`
      text-decoration: none;
      color: black;
    h4 {
        margin-bottom: 0;        
    }
    div {
       font-size: .8em;
    }
`
};

const StyledSimplePostListing = styled(SimplePostListing)`  
  ul {
    list-style: none;
    margin-left: 0;
    padding-left: 0;
    li {
        margin-bottom: .25em;
        box-shadow: ${theme.boxShadow};
        padding: .7em;
        border-radius: 5px;
    }
  }
`;

export default StyledSimplePostListing;