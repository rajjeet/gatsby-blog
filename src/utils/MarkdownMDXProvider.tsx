import React, { useState } from 'react';
import { MDXProvider } from '@mdx-js/react';
import { MDXRenderer } from 'gatsby-plugin-mdx';
import { AutoLinkHeader } from '../components/auto-link-header';

type TProps = {
  content: React.ReactNode;
}

export const MarkdownMdxProvider: React.FC<TProps> = ({ content }) => {
  const [components] = useState(
    ['h1', 'h2', 'h3', 'h4', 'h5', 'h6'].reduce((a, v) => {
      a[v] = AutoLinkHeader(v); // eslint-disable-line no-param-reassign
      return a;
    }, {}),
  );

  return (
    <MDXProvider components={components}>
      {content && <MDXRenderer>{content}</MDXRenderer>}
    </MDXProvider>
  );
};
