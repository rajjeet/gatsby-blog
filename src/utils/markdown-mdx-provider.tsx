import React, { useState } from 'react';
import { MDXProvider } from '@mdx-js/react';
import { MDXRenderer } from 'gatsby-plugin-mdx';
import styled from 'styled-components';
import { AutoLinkHeader } from '../components/auto-link-header';

type TProps = {
  content: React.ReactNode;
}

const Wrapper = styled.div`
  background-color: whitesmoke;
  padding: 1rem;
  margin-bottom: 1rem;
  border-radius: ${(props): string => props.theme.borderRadius};
  box-shadow: ${(props): string => props.theme.lightBoxShadow};  
`;

const FigCaption = styled.figcaption`
  font-style: italic;
  color: dimgrey;
  margin: .5rem 0; 
  text-align: center;
  font-size: 1rem;
`;

const Paragraph = styled.div`
  margin-bottom: 1rem;
`;

const Anchor = ({ children, href, className }): JSX.Element => (
  <a className={className} href={href} target="_blank" rel="noopener noreferrer">{children}</a>
);

const StyledAnchor = styled(Anchor)`
  color: ${(props): string => props.theme.primaryColor};
  text-decoration: none;
  &:hover, &:active {
    color: ${(props): string => props.theme.secondaryColor};
  }
`;

export const MarkdownMdxProvider: React.FC<TProps> = ({ content }) => {
  const [components] = useState(
    {
      ...['h1', 'h2', 'h3', 'h4', 'h5', 'h6'].reduce((a, v) => {
        a[v] = AutoLinkHeader(v); // eslint-disable-line no-param-reassign
        return a;
      }, {}),
      figcaption: FigCaption,
      wrapper: Wrapper,
      p: Paragraph,
      a: StyledAnchor,
    },
  );

  return (
    <MDXProvider components={components}>
      {content && <MDXRenderer>{content}</MDXRenderer>}
    </MDXProvider>
  );
};
