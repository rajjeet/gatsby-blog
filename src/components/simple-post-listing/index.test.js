import React from 'react';
import renderer from 'react-test-renderer';
import SimplePostListing from './index';
import { createMockPosts } from '../../utils/testing';

const makeProps = () => ({
  posts: createMockPosts,
});

describe('<SimplePostListing />', () => {
  it('should render', () => {
    const tree = renderer.create(<SimplePostListing {...makeProps()} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
