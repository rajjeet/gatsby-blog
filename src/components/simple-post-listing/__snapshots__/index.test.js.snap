// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SimplePostListing /> should render 1`] = `
<DocumentFragment>
  <div>
    <h3>
      Blog Posts
    </h3>
    <ul
      class="sc-bwzfXH dVzAqp"
    >
      <a
        aria-label="See post: Adding a Project Section to My Website"
        class="sc-bdVaJa jpiBPE"
        href="/adding-project-section-to-website/"
      >
        <li>
          <h4>
            Adding a Project Section to My Website
          </h4>
          <div>
            <span>
              11 May, 2019
            </span>
            <span>
              2 mins read
            </span>
          </div>
        </li>
      </a>
      <a
        aria-label="See post: State Management using React Hooks"
        class="sc-bdVaJa jpiBPE"
        href="/state-management-react-hooks/"
      >
        <li>
          <h4>
            State Management using React Hooks
          </h4>
          <div>
            <span>
              02 May, 2019
            </span>
            <span>
              4 mins read
            </span>
          </div>
        </li>
      </a>
    </ul>
  </div>
</DocumentFragment>
`;
